include/albatross/src/covariance_functions/radial.hpp

```diff
@@ -59,6 +59,28 @@ class SquaredExponential
     return "squared_exponential[" + this->distance_metric_.get_name() + "]";
   }
 
+  std::vector<double> _ssr_impl(const std::vector<double> &xs) const {
+    double min = *std::min_element(xs.begin(), xs.end());
+    double max = *std::max_element(xs.begin(), xs.end());
+
+    double range = max - min;
+
+    // using 1/10th of the length scale should result in grids with
+    // one percent decorrelation between them.
+    double n = ceil(10 * range / squared_exponential_length_scale.value);
+    n = std::max(n, 3.);
+    double step = range / n;
+
+    std::vector<double> output;
+    double x = min;
+    while (x < max) {
+      output.push_back(x);
+      x += step;
+    }
+    output.push_back(max);
```

Comment:
I would loop this over `n` instead of a while. I think it makes it easier to read and allows the addition of a number close to `max` to be added in the loop.

edit: or use the `linspace` function moved in this commit.

Reply:
Ha, yeah forgot to straighten that all out.  I used the linspace method first (that's why I moved it) but that required casting `n` to a `std::size_t` which probably isn't much of an issue (we do it elsewhere) but I do think there is an advantage to avoiding that as it could cause overflow which is how I landed where it currently is.